src/Control/Carrier/Telemetry/Utils.hs

```diff
@@ -86,7 +90,33 @@ getCurrentCliEnvironment =
     else CliProductionEnvironment
 
 getCommandArgs :: IO [Text]
-getCommandArgs = map toText <$> Environment.getFullArgs
+getCommandArgs = redactApiKeyFromCmdArgs <$> (map toText <$> Environment.getFullArgs)
```

Comment:
[nit]: Avoid double-`fmap` over the same functor, prefer combining the `fmap`-ped operations.
This has performance implications, but in this case the difference is negligible.
```suggestion
getCommandArgs = redactApiKeyFromCmdArgs . map toText <$> Environment.getFullArgs
```